### PR TITLE
Feature/bccat 5 pgp signature

### PR DESCRIPTION
**Pull request by Igor-Ivaniuk**
PGP route and fix for default properties

### CODE REVIEW

File: catalyst-generator/src/main/resources/templates/camel/router/PGPRouter.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import org.springframework.stereotype.Component;
+
+/**
+ * Sample Camel route to demonstrate encryption and decryption using PGP
+ */
+@Component
+public class PGPRouter extends RouteBuilder {
+
+  @Value("${catalyst.pgp.privateKeyFile}")
+  private String privateKeyFile;
```
Comment (oborovyk) [inline catalyst-generator/src/main/resources/templates/camel/router/PGPRouter.java]: Better to use Property class like in sftp sample with constructor property injection.

File: catalyst-generator/src/main/java/com/intellecteu/catalyst/generator/ProjectGenerator.java [hunk 2]
```diff
@@ -342,19 +346,25 @@ private void appendRouter(String usecaseName, File srcDir, Map<String, Object> m
...
   }
 
+  private void appendDefaultProperties(StringBuilder appProperties) throws IOException {
+    try (InputStream defaultPropsFile = ResourceUtils
+        .getURL("classpath:templates/camel/properties/default.yml").openStream();) {
+      String defaultProperties = CharStreams.toString(
+          new InputStreamReader(defaultPropsFile, Charsets.UTF_8));
+      appProperties.append(defaultProperties).append(System.lineSeparator());
+    } catch (FileNotFoundException ex) {
+      log.warn("Default property file not found: {}", ex.getMessage());
```
Comment (oborovyk) [inline catalyst-generator/src/main/java/com/intellecteu/catalyst/generator/ProjectGenerator.java]: In such case better to let the app crash. Default properties should not be missing

**Review by oborovyk — CHANGES_REQUESTED** [review]

File: catalyst-generator/src/main/java/com/intellecteu/catalyst/generator/ProjectGenerator.java (hunk 2 above)
Comment (Igor-Ivaniuk) [reply catalyst-generator/src/main/java/com/intellecteu/catalyst/generator/ProjectGenerator.java]: Fixed

File: catalyst-generator/src/main/resources/templates/camel/router/PGPRouter.java (hunk 1 above)
Comment (Igor-Ivaniuk) [reply catalyst-generator/src/main/resources/templates/camel/router/PGPRouter.java]: IMO, for simple string-based Camel properties it is okay to embed them into route definition, ti keep everything simple

File: catalyst-generator/src/main/java/com/intellecteu/catalyst/generator/ProjectGenerator.java [hunk 3]
```diff
@@ -342,19 +346,23 @@ private void appendRouter(String usecaseName, File srcDir, Map<String, Object> m
     }
   }
 
+  private void appendDefaultProperties(StringBuilder appProperties) throws IOException {
+    String defaultProperties = CharStreams.toString(
```
Comment (oborovyk) [inline catalyst-generator/src/main/java/com/intellecteu/catalyst/generator/ProjectGenerator.java]: use try with resources without exception handling

**Review by oborovyk — APPROVED** [review]

**Pull request merged**